lib/_http_server.js

```diff
@@ -485,8 +485,10 @@ function connectionListener(socket) {
       res.assignSocket(socket);
     }
 
+    socket.setTimeout(self.timeout);
```

Comment:
I changed my mind after considering following cases:
In this place we want to renew ```timeout```  but we may have 4 situations:
1. ```timeout``` is 0 and ```keepAliveTimeout``` is also 0, so we need no action here;
2. ```timeout``` is not 0 and ```keepAliveTimeout``` is 0, we also need no action, because current socket timeout remains ```server.timeout```;
3. ```timeout``` is 0 and ```keepAliveTimeout``` is not 0, in this case we need to ```unenroll``` timer for timeout event and we can do this by ```socket.setTimeout(0)``` or by ```socket.setTimeout(self.timeout)``` because ```self.timeout``` is 0 and ```Socket.prototype.setTimeout(msecs, callback)``` with single parameter ```msecs === 0``` do what we want;
4. ```timeout``` is not 0 and ```keepAliveTimeout``` is not 0, we need ```socket.setTimeout(self.timeout)```;

So I propose:
```JavaScript
if (self.keepAliveTimeout) socket.setTimeout(self.timeout);
```